### DISCUSSION TITLE
Single tenant database

### DISCUSSION THREAD

**Discussion by Ngomana** [sl-nx/superlogin-next#37] [General]
I'm curios, every time I register a user, it creates a separate database for them and would like to know why, I'm coming from mongodb and everything was shared, to give the user there data we used a `partition key` with the `user id` to get there specific data. Is it possible to achieve the following with couchdb?

**Comment by fynnlyte** [comment]
https://www.joshmorony.com/creating-a-multiple-user-app-with-pouchdb-couchdb/ -> here the rationale for the DB per user approach is described.

It definitely has quite some limitations (no easy querying across users, performance suffers when you have too many DBs), but if you want an offline-first app that syncs directly via PouchDB/rxdb, it can be a good choice.

But you could also just use `couch-auth` for user management and go for a classical approach where a NodeJS API handles all requests made by the client.

**Reply by Ngomana** [reply]
page doesn't load

**Reply by Ngomana** [reply]
Thank you, I'm sold... I might go for `per user database`
